src/server.rs

```diff
@@ -1,14 +1,16 @@
 use std::collections::BTreeMap;
 use std::iter;
-use std::time::Duration;
+use std::sync::Arc;
 
-use redis::Pipeline;
+use anyhow::Result;
+use redis::{aio::ConnectionManager, Pipeline};
 
 use rand::{Rng, distributions};
 
-use threadpool::ThreadPool;
+use async_channel::{bounded, Receiver, Sender};
```

Comment:
Tokio provides a `mpsc` channel, but the code previously relied on using a mpmc channel (to allow cloning) so I added this crate. https://docs.rs/async-channel/latest/async_channel/